### PR TITLE
Feat: update workflow to use ruff

### PR DESCRIPTION
**Pull request by benyissa**
## Changes Made

In this pull request, we have made the decision to switch from pylint and black to using Ruff exclusively. The rationale for this change is as follows:

### Ruff
[Ruff](https://docs.astral.sh/ruff/), an extremely fast Python linter and code formatter written in Rust, offers a robust solution for enforcing coding standards and identifying potential issues. Its comprehensive rule set and customizable configurations, along with seamless integration with popular code editors, provide real-time feedback, enabling developers to catch errors as they code. Ruff is black compatible and functions as both a linter and code formatter.

### Pylint (Previous Configuration)
Pylint, known for its extensive AST analysis and modular system, is a powerful tool. However, its drawback is its excruciating slowness, which impacts the development flow.

## Why the Change?

The decision to move to Ruff exclusively was driven by the desire for a faster and more immediate feedback loop during development. Ruff's speed and real-time feedback align with our goal of providing developers with quick insights into potential issues as they write code. This change aims to speed up checks and enhance the development flow.

### CODE REVIEW

File: tests/test-requirement.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 pylint
```
Comment (deadly-panda) [inline tests/test-requirement.txt:1]:
```suggestion
```

File: tests/test-requirement.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 pytest
-black
+ruff
```
Comment (mohsinenar) [inline tests/test-requirement.txt:3]: remove pylint too

File: tests/test-requirement.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
 mypy
```
Comment (PiranhaSa) [inline tests/test-requirement.txt:4]: does ruff replace mypy and pylint ? 
```suggestion
mypy
```

**Review by mohsinenar — DISMISSED** [review]

File: .github/workflows/code_quality_check.yml [hunk 4]
```diff
@@ -32,19 +27,21 @@ jobs:
...
+
+      - name: Install Dependencies
         run: |
           python -m pip install --upgrade pip
           python -m pip install -r requirement.txt
           python -m pip install -r tests/test-requirement.txt
-      - name: Running Black.
-        uses: psf/black@stable
+
+      - name: Run Ruff Formatter (Check)
```
Comment (3asm) [inline .github/workflows/code_quality_check.yml:37]: I would rather have the name denote the action, not the tool.

```suggestion
      - name: Checking code formatting.
```

File: .github/workflows/code_quality_check.yml [hunk 5]
```diff
@@ -32,19 +27,21 @@ jobs:
...
         with:
-          options: "--check"
-      - name: Running linter.
-        run: |
-          pylint --rcfile=.pylintrc --ignore=tests/ agent/
-          pylint --rcfile=.pylintrc -d C0103,W0613 tests/
-      - name: Running static types checker.
+          args: format --check
+
+      - name: Run Ruff Linter
```
Comment (3asm) [inline .github/workflows/code_quality_check.yml:42]:
```suggestion
      - name: Running linter.
```

File: .github/workflows/code_quality_check.yml [hunk 6]
```diff
@@ -32,19 +27,21 @@ jobs:
...
+        uses: chartboost/ruff-action@v1
+
+      - name: Run Mypy (Static Type Checker)
```
Comment (3asm) [inline .github/workflows/code_quality_check.yml:45]:
```suggestion
      - name: Running static types checker.
```

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**